plugins/attachment-impl/src/index.ts

```diff
@@ -0,0 +1,135 @@
+import attachment from '@anticrm/attachment'
+import { Ref, Space } from '@anticrm/core'
+import { getMetadata } from '@anticrm/platform'
+import { PlatformError, Status, Severity } from '@anticrm/status'
+import pluginCore from '@anticrm/plugin-core'
+
+export { default as Attachments } from './components/Attachments.svelte'
+
+export default async (): Promise<AttachmentService> => {
+  const fileServerURL = getMetadata(attachment.metadata.FilesUrl) ?? ''
+  if (fileServerURL === '') {
+    throw new PlatformError(new Status(Severity.ERROR, attachment.status.NoFileServerUri, {}))
+  }
+  const token = getMetadata(pluginCore.metadata.Token)
+
+  async function upload (
+    file: File,
+    key: string,
+    space: Ref<Space>,
+    progressCallback?: (progress: number) => void
+  ): Promise<void> {
+    const params = {
+      key,
+      space,
+      type: file.type
+    }
+    const req = await fetch(fileServerURL + 'file', {
+      method: 'PUT',
+      credentials: 'include',
+      headers: {
+        'Content-Type': 'application/json;charset=utf-8',
+        Token: token
+      },
+      body: JSON.stringify(params)
+    })
+    const url = await req.text()
+    await uploadToServer(url, file, progressCallback)
+  }
+
+  async function uploadToServer (
+    url: string,
+    file: File,
+    progressCallback?: (progress: number) => void
+  ): Promise<XMLHttpRequest> {
+    return await new Promise(function (resolve, reject) {
+      const xhr = new XMLHttpRequest()
+
+      xhr.onreadystatechange = () => {
+        if (xhr.readyState === 4) {
+          if (xhr.status === 200) {
+            resolve(xhr)
+          } else {
+            reject(xhr)
+          }
+        }
+      }
+
+      if (progressCallback != null) {
+        xhr.upload.onprogress = (e) => {
+          if (e.lengthComputable) {
+            const percentComplete = (e.loaded / file.size) * 100
+            progressCallback(percentComplete)
+          }
+        }
+      }
+
+      xhr.open('PUT', url)
+      xhr.send(file)
+    })
+  }
+
+  async function remove (key: string, space: Ref<Space>): Promise<void> {
+    const params = {
+      key,
+      space
+    }
+    await fetch(fileServerURL + 'file', {
+      method: 'DELETE',
+      headers: {
+        'Content-Type': 'application/json;charset=utf-8',
+        Token: token
+      },
+      body: JSON.stringify(params)
+    })
+  }
+
+  function generateLink (key: string, space: Ref<Space>, name: string, format: string): string {
+    const regex = RegExp(`${format}$`)
+    const downloadName = regex.test(name) ? name : name + '.' + format
+    return `${fileServerURL}file/${space}/${key}/${downloadName}`
+  }
+
+  async function authorizeFileServer (token: string): Promise<void> {
+    await fetch(fileServerURL, {
+      method: 'POST',
+      credentials: 'include',
+      headers: {
+        'Content-Type': 'application/json;charset=utf-8',
+        Token: token
```

Comment:
Could we use https://developer.mozilla.org/ru/docs/Web/API/Document/cookie instead?